File: backend/app/schemas/module_output.py
from typing import Literal, Any
from pydantic import BaseModel, Field


class Finding(BaseModel):
    title: str
    description: str
    evidence: dict[str, Any] = Field(default_factory=dict)
    recommendation: str
    severity: Literal["low", "medium", "high"] = "low"


class ModuleOutput(BaseModel):
    module: str
    status: Literal["success", "error"]
    score: int = Field(ge=0, le=100)
    severity: Literal["low", "medium", "high"]
    findings: list[Finding] = Field(default_factory=list)
    artifacts: list[str] = Field(default_factory=list)
    summary: str = ""

File: frontend/src/types/moduleOutput.ts
export type Severity = "low" | "medium" | "high";

export interface Finding {
  title: string;
  description: string;
  evidence: Record<string, any>;
  recommendation: string;
  severity: Severity;
}

export interface ModuleOutput {
  module: string;
  status: "success" | "error";
  score: number;
  severity: Severity;
  findings: Finding[];
  artifacts: string[];
  summary: string;
}

export interface Project {
  id: number;
  name: string;
  description: string;
  target_column?: string;
  created_at: string;
}
